### PR TITLE
Initializers for sync/single

### PR DESCRIPTION
**Pull request by benharsh**
This PR updates sync and single types to use initializers. Note that these types rely on ``chpl__initCopy`` returning a different type, so this isn't a total conversion to pure initializers.

Two .bad files are removed in order to reduce overhead of updating them as the initializer work progresses.

A breakpoint has been added for ``resolveInitVar`` to aid debugging.

Testing:
- [x] full local

### CODE REVIEW

**Comment by benharsh** [comment]
@lydia-duncan : would you mind reviewing?

File: modules/internal/ChapelSyncvar.chpl [hunk 1]
```diff
@@ -111,18 +111,30 @@ module ChapelSyncvar {
...
-    }
-
-    proc _syncvar(type valType, const other : _syncvar(valType)) {
-      ensureFEType(valType);
-
-      wrapped = other.wrapped;
-      isOwned = false;
+      this.valType = valType;
+      this.wrapped = new (getSyncClassType(valType))(valType);
+      this.isOwned = true;
```
Comment (lydia-duncan) [inline modules/internal/ChapelSyncvar.chpl]: You could omit the initialization of isOwned, I think

File: modules/internal/ChapelSyncvar.chpl [hunk 2]
```diff
@@ -613,18 +629,30 @@ module ChapelSyncvar {
     var  wrapped : _singlecls(valType) = nil;
     var  isOwned : bool                = true;
 
-    proc _singlevar(type valType) {
+    proc init(type valType) {
       ensureFEType(valType);
-
+      this.valType = valType;
       wrapped = new _singlecls(valType);
       isOwned = true;
```
Comment (lydia-duncan) [inline modules/internal/ChapelSyncvar.chpl]: Could also omit this one

File: modules/internal/ChapelSyncvar.chpl (hunk 1 above)
Comment (benharsh) [reply modules/internal/ChapelSyncvar.chpl]: Omit because of default initialization?

Comment (lydia-duncan) [reply modules/internal/ChapelSyncvar.chpl]: Yeah.  This is the same as the declared default value for the field, so omitted initialization would just insert the same line

**Comment by benharsh** [comment]
Made the change and ran a 100 or so tests to make sure nothing blew up. It all looks good, thanks for the review!

**Pull request merged**